src/bin/icgt.rs

```diff
@@ -507,39 +510,41 @@ pub async fn do_event_loop(cfg: ConnectConfig) -> Result<(), IcgtError> {
         p
     };
     'running: loop {
-        let system_event = event_pump.wait_event();
-        let event = translate_system_event(&system_event);
-        let event = match event {
-            None => continue 'running,
-            Some(event) => event,
-        };
-        trace!("SDL event_pump.wait_event() => {:?}", &system_event);
-        // catch window resize event: redraw and loop:
-        match event {
-            event::Event::Quit => {
-                debug!("Quit");
-                return Ok(());
-            }
-            event::Event::WindowSizeChange(new_dim) => {
-                debug!("WindowSizeChange {:?}", new_dim);
-                let rr: render::Result =
-                    server_call(&cfg, ServerCall::WindowSizeChange(new_dim.clone())).await?;
-                window_dim = new_dim;
-                redraw(&mut canvas, &window_dim, &rr).await?;
-            }
-            event::Event::KeyUp(ref ke_info) => debug!("KeyUp {:?}", ke_info.key),
-            event::Event::KeyDown(ref ke_info) => {
-                debug!("KeyDown {:?}", ke_info.key);
-                q_key_infos.push(ke_info.clone());
-                let rr: render::Result = {
-                    let mut buffer = u_key_infos.clone();
-                    buffer.append(&mut (q_key_infos.clone()));
-                    let rr = server_call(&cfg, ServerCall::QueryKeyDown(buffer.clone())).await?;
-                    rr
-                };
-                redraw(&mut canvas, &window_dim, &rr).await?;
+        if let Some(system_event) = event_pump.wait_event_timeout(13) {
```

Comment:
The use of `_timeout` (and a small number of milliseconds) helps a lot here in the interactive experience, since it means that every 13ms we re-check the status of the last update call.